pages/api/team-member/get-by-key.tsx

```diff
@@ -0,0 +1,14 @@
+import prisma from '../../../lib/prisma';
+
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+  // Get the key-value pair from the URL
+  const key = req.query.key.toString();
```

Comment:
this should be available only if logged in. Also, why don't we use getServerSideProps for this part of the logic?